lib/collection/cookie.js

```diff
@@ -271,7 +271,10 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
 
         if (this.extensions) {
             this.extensions.forEach(function (ext) {
-                str += '; ' + ext;
+                if (!ext.key || !ext.value) {
+                    return;
+                }
+                str += '; ' + ext.key + '=' + ext.value;
```

Comment:
Should we also type-check the `key` and `value` in this case?
And should we iterate over the whole array in  the constructor itself for this?

As far as I could see, `extensions` are only used in this method, and we're already iterating over it. Adding the check here itself might help us avoid the overhead of another iteration in the constructor.

But it would become non-DRY if some other method starts using extensions like this.

@codenirvana suggestions?

Reply:
We don't mutate user data (if set incorrectly) in the constructor function (with few exceptions) and handle them in methods like `toString` etc.